### PR TITLE
Type issue

### PR DESCRIPTION
**Pull request by FR13ndSDP**
## Checklist

The proposed changes:
- [x] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [x] include documentation in the code and/or rst files, if appropriate

- The type of `use_gpu_aware_mpi` is not consistent with documentation. 
- If HIP version is lower than 5.2, provide an option to assign the value of `num_device_partners` directly.

### CODE REVIEW

File: Src/Base/AMReX_ParallelDescriptor.H [hunk 1]
```diff
@@ -107,7 +107,7 @@ while ( false )
     void Initialize ();
     void Finalize ();
 
-    extern AMREX_EXPORT int use_gpu_aware_mpi;
+    extern AMREX_EXPORT bool use_gpu_aware_mpi;
```
Comment (WeiqunZhang) [inline Src/Base/AMReX_ParallelDescriptor.H:110]: Casting from int to bool is well defined. So I don't think we need to make any changes here. Is your compiler complaining about it?

Comment (FR13ndSDP) [reply Src/Base/AMReX_ParallelDescriptor.H:110]: Actually, there is no problem compiling the code, but if I use "amrex.use_gpu_aware_mpi=False" according to the docs, there will be a parmparse error.

Comment (FR13ndSDP) [reply Src/Base/AMReX_ParallelDescriptor.H:110]: I've recently tested the GPU-aware MPI feature on a platform that uses Hygon DCU (AMD arch gfx906), and the result is not very satisfying.

Comment (WeiqunZhang) [reply Src/Base/AMReX_ParallelDescriptor.H:110]: The documentation is incorrect. It should be 0 or 1.

Comment (WeiqunZhang) [reply Src/Base/AMReX_ParallelDescriptor.H:110]: Even if the type is bool, the flag needs to be `false`, not `False`. So I will fix the documentation. If we change the type here to bool, we need to do that to its definition in AMReX_ParallelDescriptor.cpp too.

Comment (WeiqunZhang) [reply Src/Base/AMReX_ParallelDescriptor.H:110]: #3562

Comment (WeiqunZhang) [reply Src/Base/AMReX_ParallelDescriptor.H:110]: By the way, using `1` or `0` is always right for ParmParse boolean type parameters whether the C++ type is `int` or `bool`.

Comment (FR13ndSDP) [reply Src/Base/AMReX_ParallelDescriptor.H:110]: Yeah, ok. So we just need to correct the documentation.

**Comment by FR13ndSDP** [comment]
I will close this draft because more consideration is needed for HIP <= 5

**Pull request closed without merging**